src/model_bias_analysis.py

```diff
@@ -50,19 +63,30 @@ def read_identity_terms(identity_terms_path):
     with open(identity_terms_path) as f:
         return [term.strip() for term in f.readlines()]
 
-
-def balanced_term_subset(df, term, text_col):
-    """Returns data subset containing term balanced with sample of other data.
+    
+def add_subgroup_columns_from_text(df, text_column, subgroups):
+    """Adds a boolean column for each subgroup to the data frame.
+    
+    New column contains True if the text contains that subgroup term.
+    """
+    for term in subgroups:
+        df[term] = df[text_column].apply(lambda x: term in x)
```

Comment:
I think this matching behavior isn't the same as the original code:
1. it's case-sensitive
2. it matches any substring, not just word-boundaries (so "blacksmith" and "whitewash" would start matching)

This probably doesn't cause any changes in the madlibs dataset, but probably worth having this code be general-purpose?

Open to changing the behavior, just want to make sure it's an intentional change?

For reference:
new: `df[text_column].apply(lambda x: term in x)`
old: `df[text_col].str.contains(r'\b{}\b'.format(term), case=False)`

Reply:
Ah great catch! I meant to go back and fix this, but totally forgot :-/ Fixed now.